entity-service/src/test/kotlin/com/egm/stellio/entity/authorization/Neo4jAuthorizationRepositoryTest.kt

```diff
@@ -169,6 +171,54 @@ class Neo4jAuthorizationRepositoryTest {
         neo4jRepository.deleteEntity(apiaryUri)
     }
 
+    @Test
+    fun `it should filter entities authorized per a specific access policy`() {
+        createEntity(
+            apiaryUri, listOf("Apiary"),
+            mutableListOf(
+                Property(
+                    name = EGM_SPECIFIC_ACCESS_POLICY,
+                    value = SpecificAccessPolicy.AUTH_READ.name
+                )
+            )
+        )
+        createEntity(apiary02Uri, listOf("Apiary"), mutableListOf())
+
+        val authorizedEntities =
+            neo4jAuthorizationRepository.filterEntitiesWithSpecificAccessPolicy(
+                listOf(apiaryUri, apiary02Uri),
+                listOf(SpecificAccessPolicy.AUTH_READ.name)
+            )
+
+        assert(authorizedEntities == listOf(apiaryUri))
+
+        neo4jRepository.deleteEntity(apiaryUri)
+        neo4jRepository.deleteEntity(apiary02Uri)
+    }
+
+    @Test
+    fun `it should filter entities with auth write if auth read and auth write are asked for`() {
```

Comment:
I think that it could be interesting here to create the apiary02 with AUTH_READ access policy and check that only the apiary with the WRITE access policy is returned.

Reply:
done